File: C++/freeCodeCamp.org_C++ Tutorial for Beginners/src/018_Building a Guessing Game/BuildingAGuessingGame.cpp
#include <iostream>

/* Simple number guessing game */

using namespace std;

int main()
{
    int secretNumber = 7;
    int guess;
    int guessCount = 0;
    int guessLimit = 3;
    bool outOfGuesses = false;

    do
    {
        if(guessCount < guessLimit)
        {
            cout << "Enter guess: ";
            cin >> guess;
            guessCount++;
        } else {
            outOfGuesses = true;
            break;
        }

        if(guess != secretNumber)
        {
            cout << endl << "Your Guess was not correct" << endl;
            cout << "Guesses remaining: " << (guessLimit - guessCount) << endl << endl;
        }
        
    }while(guess != secretNumber && !outOfGuesses);

    if(outOfGuesses)
    {
        cout << "Out of guesses. You lose" << endl;
    } else {
        cout << endl << "Your guess was correct" << endl;
    }

    return 0;
}

File: Python/002_OOP Basics/018_Building a Guessing Game/BuildingAGuessingGame.py
guess = 0
secretNumber = 7
guessCount = 0
guessLimit = 3
outOfGuesses = False

while True:
    if(guessCount < guessLimit):
        guess = int(input("Enter guess: "))
        guessCount += 1
    else:
        outOfGuesses = True
        continue

    if(guess != secretNumber):
        print("\nYour Guess was not correct")
        print("Guesses remaining: ", guessLimit - guessCount, "\n")
    
    if(guess != secretNumber and not outOfGuesses):
        continue

    if(outOfGuesses):
        print("Out of guesses. You lose")
        break
    else:
        print("Your guess was correct")
        break